src/plugin.js

```diff
@@ -25,20 +26,42 @@ const onPlayerReady = (player, options) => {
   player.addClass('vjs-download-button');
 
   if (!player.controlBar.childNameIndex_.hasOwnProperty('DownloadButton')) {
-    let linkProps = {
-      className: buildCSSClass(),
-      href: player.currentSrc(),
-      title: options.text,
-      download: ''
-    };
-    let linkAttrs = {
-      'aria-live': 'polite',
-      'aria-label': options.text
-    };
 
-    player.controlBar.addChild(new ClickableComponent(this, {
-      el: ClickableComponent.prototype.createEl('a', linkProps, linkAttrs)
-    }));
+    if (options.allDownloads) {
+      let sources = player.el_.querySelectorAll('source');
```

Comment:
why not [currentSources()](http://docs.videojs.com/Player.html#currentSources)?

Reply:
Aha. Hmm, I did not know of that function. I'll redo and submit a new PR.